src/classes/Producer.js

```diff
@@ -55,16 +85,51 @@ class Producer {
         await this.connection.setLocalDescription(answer);
 
         console.log(
-          `Sending ${this.connection.localDescription.type} to ${this.id}`
+          `Sending ${this.connection.localDescription.type} to ${this.clientId}`
         );
-        this.socket.emit('producerHandshake', {
-          description: this.connection.localDescription,
-          clientId: this.id,
-        });
+
+        const payload = {
+          action: 'handshake',
+          data: {
+            type: 'producer',
+            sender: this.sfuId,
+            receiver: this.clientId,
+            description: this.connection.localDescription,
+          },
+        };
+
+        console.log('Sending producer answer');
+        console.log('Receiver is:', this.clientId);
+
+        this.socket.send(JSON.stringify(payload));
+        this.processQueuedCandidates();
       }
     } else if (candidate) {
       try {
-        console.log('Adding an ice candidate');
+        this.handleReceivedIceCandidate(candidate);
+      } catch (e) {
+        if (candidate.candidate.length > 1) {
+          console.log('unable to add ICE candidate for peer', e);
+        }
+      }
+    }
+  }
+
+  async handleReceivedIceCandidate(candidate) {
```

Comment:
this is to deal with network latency, right?

Reply:
Exactly. If an attempt is made to add ICE Candidates before the RemoteDescription is set, it will fail outright. This will handle that situation and just stored the candidates until the Offer/Answer is received and then the RemoteDescription is set, at which point it will process all of the candidates that have been stored and add them appropriately.